lms/static/coffee/src/instructor_dashboard/membership.coffee

```diff
@@ -670,6 +670,564 @@ class AuthList
+      if selected[1].text == "Section"
+        @arr = [{'text':selected[0].text, 'id':selected[0].id},
+                {'text':selected[1].text, 'id':selected[1].id},
+                {'text':selected[4].text, 'id':selected[4].id},
+                {'text':selected[5].text, 'id':selected[5].id}]
+        @arr_text = [selected[0].text, selected[1].text,
+                     selected[4].text, selected[5].text]
+
+      else
+        @arr = [{'text':selected[0].text, 'id':selected[0].id},
+                {'text':selected[1].text, 'id':selected[1].id},
+                {'text':selected[2].text, 'id':selected[2].id},
+                {'text':selected[3].text, 'id':selected[3].id}]
+        @arr_text = [selected[0].text, selected[1].text,
+                     selected[2].text, selected[3].text]
+
+      for thing in @arr
+        if thing['text'] == ""
+          $("#incompleteMessage")[0].innerHTML =
+            "Query is incomplete. Please make all the selections."
+          return
+
+      $("#incompleteMessage")[0].innerHTML = ""
+      @chosen = selected[0].text
+      @tr = @start_row(@chosen.toLowerCase(), @arr, "", $("#queryTableBody"))
+      @use_query_endpoint = @$query_endpoint + "/" + @arr_text.slice(0,2).
+        join("/") + "/" + @arr[2].id
+      @filtering = @arr[3].text
+      @entityName = @arr[2].text
+      @reload_students(@tr)
+      @$email_list_containers.find('select.single-email-selector').
+        prop('selectedIndex', 0)
+      $(".problem_specific").removeClass('active')
+      $(".section_specific").removeClass('active')
+
+  get_saved_temp_queries: (cb)->
+    $.ajax
+      dataType: 'json'
+      url: @$temp_queries_endpoint
+      data: rolename: 'instructor'
+      success: (data) -> cb? null, data
+      error: std_ajax_err ->
+        cb? gettext("Error getting saved temp queries")
+
+  #get a user's in-progress queries and load them into active queries
+  load_saved_temp_queries: ->
+    @get_saved_temp_queries (error, data) =>
+      # abort on error
+      return @show_errors error unless error is null
+      $("#queryTableBody tr").remove()
+      queries = data['queries']
+      # use _.each instead of 'for' so that member
+      # is bound in the button callback.
+      _.each queries, (query) =>
+        query_id = query['id']
+        block_id = query['block_id']
+        block_type = query['block_type']
+        state_key = block_type + "/" + block_id
+        display_name = query['display_name']
+        display_entity = {'text':display_name, 'id':state_key}
+        filter_on = {'text':query['filter_on']}
+        inclusion = {'text':query['inclusion']}
+        done = query['done']
+        type = {'text':query['type']}
+        arr = [inclusion, type, display_entity, filter_on, done]
+        @tr = @start_row(inclusion['text'].toLowerCase(),arr,
+          {'class':["working"],'query':query_id},  $("#queryTableBody"))
+        @check_done()
+
+
+  get_saved_queries: (cb)->
```

Comment:
This method doesn't appear to use any state members/methods, so it doesn't need to be defined as an instance method.